CI/travis/build_deploy_doc.sh

```diff
@@ -44,10 +44,10 @@ echo_green "Documentation was generated successfully!"
 # If the current build is not a pull request and it is on master the 
 # documentation will be pushed to the gh-pages branch
 ############################################################################
-if [[ "${TRAVIS_PULL_REQUEST}" == "false" && "${TRAVIS_BRANCH}" == "master" ]]
+if [[ "${SYSTEM_PULLREQUEST_PULLREQUESTNUMBER}" == "false" && "${SYSTEM_PULLREQUEST_TARGETBRANCH}" == "master" ]]
```

Comment:
I don't think these 2 variables hold the expected values.
On Travis CI:
TRAVIS_PULL_REQUEST: The pull request number if the current job is a pull request, “false” if it’s not a pull request.
TRAVIS_BRANCH: for push builds, or builds not triggered by a pull request, this is the name of the branch.

The vars for Azure Pipelines are not exactly the equivalent, and this might cause issues when the documentation is pushed upstream.
Maybe there is something in the predefined variables that can check if the build reason is a PR, for the first one.
While for the second one, we need the current build branch. Right now I think we are checking which branch is the target of this PR.